usagov/vote-gov-drupal
# VOTE-441: Add husky support. Include git hooks for git-secrets to pre…

…vent committing AWS secrets.



## Jira ticket

[VOTE-441](https://cm-jira.usa.gov/browse/VOTE-441)

## Description

Install husky and git-secrets to enable the use of commit hooks to prevent AWS secrets from being committed.

## Deployment and testing

### Pre-deploy

1. Insert any required steps to take before merging and running post-deploy.

### Post-deploy

1. Insert any required steps to take before beginning test.

### QA/Test

1. Ensure that you npm installed locally, outside of any Docker container.
2. Install git-secrets locally - https://github.com/awslabs/git-secrets#installing-git-secrets
3. Confirm `git-secrets` executes.
4. Once installed, run `npm install` which will install Husky for the project.
5. Confirm that you receive the message `husky - Git hooks installed` after Step 3.
6. Create or alter a file to include a random AWS key. An example key is available here: https://awsteele.com/blog/2020/09/26/aws-access-key-format.html
Example: demo.txt with key, ASIA...................A
7. Run the following
`git add {test_file.txt}`
`git commit -m"I should not see this in logs."` - The commit should be blocked and you should receive a message showing where a secret was found
8. Use `git restore --staged <file>` to unstage your changes so you can delete the test key.

TIP: You can always issue the following command for a scan
`git secrets --scan` - You should receive a message showing where a secret was found

## Checklist for the Developer




## Checklist for the Peer Reviewers
